File: mobile/ios/PortaliaCore/Sources/PortaliaCore/Models/StaffModels.swift
import Foundation

public struct StaffLoginResponse: Decodable {
    public let accessToken: String
    public let mustChangePassword: Bool
    public let csrfToken: String
}

/// POST /auth/change-password devolve só isto — sem `mustChangePassword`
/// (o chamador sabe que virou false só por ter tido sucesso na troca).
public struct StaffChangePasswordResponse: Decodable {
    public let accessToken: String
    public let csrfToken: String
}

public struct StaffRefreshResponse: Decodable {
    public let accessToken: String
    public let csrfToken: String
    public let mustChangePassword: Bool
}

public struct StaffMeResponse: Decodable {
    public let id: String
    public let email: String
    public let role: String
    public let resaleId: String?
    public let clientId: String?
    public let condominiumId: String?
    public let mustChangePassword: Bool
}

public struct ReportsOverview: Decodable {
    public let units: Int
    public let residents: Int
    public let vehicles: Int
    public let activeProviders: Int
    public let visitorsToday: Int
    public let pendingDeliveries: Int
    public let openOccurrences: Int
    public let keysCheckedOut: Int
    public let activePanicAlerts: Int
}

public struct UnitResponse: Decodable, Identifiable {
    public let id: String
    public let condominiumId: String
    public let identifier: String
    public let block: String?
    public let floor: String?
    public let isActive: Bool
}

public struct ExtensionResponse: Decodable, Identifiable {
    public let id: String
    public let unitId: String
    public let condominiumId: String
    public let number: String
    public let sipUsername: String
    public let isActive: Bool
}

public struct CallResponse: Decodable, Identifiable {
    public let id: String
    public let condominiumId: String
    public let externalCallId: String?
    public let callerType: String
    public let callerExtensionId: String?
    public let calleeType: String
    public let calleeExtensionId: String?
    public let status: String
    public let queuedAt: Date?
    public let ringingAt: Date?
    public let answeredAt: Date?
    public let endedAt: Date?
    public let endReason: String?
    public let createdAt: Date
}

public struct AnnouncementResponse: Decodable, Identifiable {
    public let id: String
    public let condominiumId: String
    public let title: String
    public let body: String
    public let createdBy: String
    public let createdAt: Date
    public let updatedAt: Date
}

public struct DeliveryResponse: Decodable, Identifiable {
    public let id: String
    public let unitId: String
    public let condominiumId: String
    public let residentId: String?
    public let description: String
    public let carrier: String?
    public let courierName: String?
    public let trackingCode: String?
    public let volumeCount: Int
    public let status: String
    public let pickedUpBy: String?
    public let pickupDocument: String?
    public let pickupCodeLast4: String?
    public let pickupCredentialExpiresAt: Date?
    public let receivedAt: Date
    public let collectedAt: Date?
}

/// `Identifiable` porque a tela de entregas apresenta a credencial via
/// `.sheet(item:)`, que exige isso. O id é o da própria entrega (não um
/// `UUID()` novo): um id gerado a cada decodificação faria a sheet reabrir
/// sozinha sempre que o valor fosse recriado.
public struct DeliveryCredentialResponse: Decodable, Identifiable {
    public let delivery: DeliveryResponse
    public let pickupCode: String
    public let qrToken: String
    public let qrPayload: String
    public let expiresAt: Date?

    public var id: String { delivery.id }
}

public struct PanicAlertResponse: Decodable, Identifiable {
    public let id: String
    public let condominiumId: String
    public let unitId: String?
    public let triggeredBy: String
    public let acknowledgedAt: Date?
    public let resolvedAt: Date?
    public let createdAt: Date
}

File: mobile/android/core/src/main/kotlin/com/portalia/core/model/StaffModels.kt
package com.portalia.core.model

import kotlinx.serialization.Serializable

@Serializable
data class StaffLoginResponse(val accessToken: String, val mustChangePassword: Boolean, val csrfToken: String)

@Serializable
data class StaffRefreshResponse(val accessToken: String, val csrfToken: String, val mustChangePassword: Boolean)

/** POST /auth/change-password devolve só isto — sem mustChangePassword. */
@Serializable
data class StaffChangePasswordResponse(val accessToken: String, val csrfToken: String)

@Serializable
data class StaffMeResponse(
    val id: String,
    val email: String,
    val role: String,
    val resaleId: String? = null,
    val clientId: String? = null,
    val condominiumId: String? = null,
    val mustChangePassword: Boolean,
)

@Serializable
data class ReportsOverview(
    val units: Int,
    val residents: Int,
    val vehicles: Int,
    val activeProviders: Int,
    val visitorsToday: Int,
    val pendingDeliveries: Int,
    val openOccurrences: Int,
    val keysCheckedOut: Int,
    val activePanicAlerts: Int,
)

@Serializable
data class UnitResponse(
    val id: String,
    val condominiumId: String,
    val identifier: String,
    val block: String? = null,
    val floor: String? = null,
    val isActive: Boolean,
)

@Serializable
data class ExtensionResponse(
    val id: String,
    val unitId: String,
    val condominiumId: String,
    val number: String,
    val sipUsername: String,
    val isActive: Boolean,
)

@Serializable
data class CallResponse(
    val id: String,
    val condominiumId: String,
    val externalCallId: String? = null,
    val callerType: String,
    val callerExtensionId: String? = null,
    val calleeType: String,
    val calleeExtensionId: String? = null,
    val status: String,
    val queuedAt: String? = null,
    val ringingAt: String? = null,
    val answeredAt: String? = null,
    val endedAt: String? = null,
    val endReason: String? = null,
    val createdAt: String,
)

@Serializable
data class AnnouncementResponse(
    val id: String,
    val condominiumId: String,
    val title: String,
    val body: String,
    val createdBy: String,
    val createdAt: String,
    val updatedAt: String,
)

@Serializable
data class DeliveryResponse(
    val id: String,
    val unitId: String,
    val condominiumId: String,
    val residentId: String? = null,
    val description: String,
    val carrier: String? = null,
    val courierName: String? = null,
    val trackingCode: String? = null,
    val volumeCount: Int,
    val status: String,
    val pickedUpBy: String? = null,
    val pickupDocument: String? = null,
    val pickupCodeLast4: String? = null,
    val pickupCredentialExpiresAt: String? = null,
    val receivedAt: String,
    val collectedAt: String? = null,
)

@Serializable
data class DeliveryCredentialResponse(
    val delivery: DeliveryResponse,
    val pickupCode: String,
    val qrToken: String,
    val qrPayload: String,
    val expiresAt: String? = null,
)

@Serializable
data class PanicAlertResponse(
    val id: String,
    val condominiumId: String,
    val unitId: String? = null,
    val triggeredBy: String,
    val acknowledgedAt: String? = null,
    val resolvedAt: String? = null,
    val createdAt: String,
)
